frameProcessor/src/FileWriterPlugin.cpp

```
@@ -502,42 +500,42 @@ void FileWriterPlugin::processFrame(boost::shared_ptr<Frame> frame)
-      }
+      // The frame has no subframes so write the whole frame
+      this->writeFrame(*frame);
+    }
 
-      // Check if this is a master frame (for multi dataset acquisitions)
-      // or if no master frame has been defined. If either of these conditions
-      // are true then increment the number of frames written.
-      if (masterFrame_ == "" || masterFrame_ == frame->get_dataset_name()) {
-        size_t datasetFrames = this->getDatasetFrames(frame->get_dataset_name());
-        if (datasetFrames == framesWritten_) {
-          LOG4CXX_DEBUG(logger_, "Frame " << datasetFrames << " rewritten");
-        }
-        else {
-          framesWritten_ = datasetFrames;
-        }
-        LOG4CXX_DEBUG(logger_, "Master frame processed");
+    // Check if this is a master frame (for multi dataset acquisitions)
+    // or if no master frame has been defined. If either of these conditions
+    // are true then increment the number of frames written.
+    if (currentAcquisition_.masterFrame_ == "" || currentAcquisition_.masterFrame_ == frame->get_dataset_name()) {
+      size_t datasetFrames = this->getDatasetFrames(frame->get_dataset_name());
+      if (datasetFrames == framesWritten_) {
+        LOG4CXX_DEBUG(logger_, "Frame " << datasetFrames << " rewritten");
       }
       else {
-        LOG4CXX_DEBUG(logger_, "Non-master frame processed");
+        framesWritten_ = datasetFrames;
       }
+      LOG4CXX_DEBUG(logger_, "Master frame processed");
+    }
+    else {
+      LOG4CXX_DEBUG(logger_, "Non-master frame processed");
+    }
 
-      // Check if we have written enough frames and stop
-      if (framesToWrite_ > 0 && framesWritten_ == framesToWrite_) {
-        this->stopWriting();
-      }
+    // Check if we have written enough frames and stop
+    if (currentAcquisition_.framesToWrite_ > 0 && framesWritten_ == currentAcquisition_.framesToWrite_) {
+      this->stopWriting();
+      // Start next acquisition
+      this->startWriting();
```

No, currently it will log a warning if there is no next acquisition when it tries to startWriting(). I thought this would be preferable to silently doing nothing, but maybe we could check here and log a specific warning/debug to say there was no next acquisition to start, rather than calling startWriting and hoping.